### ISSUE TITLE
client count in infoResponse is wrong (missing one)

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1028]
See the server answer in infoResponse packet:

```sh
$ qstat -d -d -d -json -q3s gg.illwieckz.net:27960
```
```sh
debug(1) 22:12:55 hcache.c:401 hcache_lookup_hostname() - looking up gg.illwieckz.net
debug(2) 22:12:55 hcache.c:412 hcache_lookup_hostname() - validating gg.illwieckz.net
debug(2) 22:12:55 hcache.c:417 hcache_lookup_hostname() - returning 3ed2472c
debug(1) 22:12:55 qstat.c:4546 bind_qserver2() - start 0x565537f91550 @ 62.210.71.44:27960
debug(1) 22:12:55 qstat.c:4696 bind_sockets() - send 62.210.71.44:27960
debug(2) 22:12:55 qstat.c:4699 bind_sockets() - calling status_query_func for 0x565537f91550 - connect
debug(3) 22:12:55 qstat.c:4774 process_func_ret() - 0x565537f91550, 0
debug(3) 22:12:55 qstat.c:4811 send_packets() - processing...
debug(2) 22:12:55 qstat.c:3118 do_work() - connected: 1
debug(2) 22:12:55 qstat.c:5741 qserver_get_timeout() - timeout for 0x565537f91550 is diff1 1000 diff2 65535
debug(2) 22:12:55 qstat.c:3138 do_work() - rc 1
debug(2) 22:12:55 qstat.c:3161 do_work() - recvfrom: 225
debug(1) 22:12:55 qstat.c:3183 do_work() - recv 376 376 62.210.71.44:27960
debug(2) 22:12:55 qstat.c:3192 do_work() - fill: 1 < 40
FROM gg.illwieckz.net:27960 len 225
    0: ffffffff 696e666f 52657370 6f6e7365  ....info Response
   16: 0a5c626f 74735c35 5c636c69 656e7473  .\bots\5 \clients
   32: 5c315c67 616d656e 616d655c 756e765c  \1\gamen ame\unv\
   48: 686f7374 6e616d65 5c5e3967 672e5e34  hostname \^9gg.^4
   64: 696c6c5e 37776965 5e31636b 7a5e392e  ill^7wie ^1ckz^9.
   80: 6e65745e 36207b20 5e356c65 20667261  net^6 {  ^5le fra
   96: 6720636f 7572746f 69735e36 207d5e39  g courto is^6 }^9
  112: 20626163 6b706f72 745e2a5c 6d61706e   backpor t^*\mapn
  128: 616d655c 70657273 6575735c 70726f74  ame\pers eus\prot
  144: 6f636f6c 5c38365c 70757265 5c315c73  ocol\86\ pure\1\s
  160: 65727665 726c6f61 645c345c 73746174  erverloa d\4\stat
  176: 735c6874 74703a2f 2f67672e 696c6c77  s\http:/ /gg.illw
  192: 6965636b 7a2e6e65 742f2373 74617473  ieckz.ne t/#stats
  208: 5c73765f 6d617863 6c69656e 74735c33  \sv_maxc lients\3
  224: 30                                   0

debug(2) 22:12:55 qstat.c:3243 do_work() - connected, pre-packet_func: 1
debug(2) 22:12:55 qstat.c:6133 deal_with_qw_packet() - deal_with_qw_packet 0x565537f91550, 225
debug(2) 22:12:55 qstat.c:6429 deal_with_q2_packet() - deal_with_q2_packet 0x565537f91550, 208
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - bots = 5
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - clients = 1
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - gamename = unv
debug(3) 22:12:55 qstat.c:7356 add_rule() - key: gamename, value: unv, flags: 1
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - hostname = ^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - mapname = perseus
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - protocol = 86
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - pure = 1
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - serverload = 4
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - stats = http://gg.illwieckz.net/#stats
debug(3) 22:12:55 qstat.c:6460 deal_with_q2_packet() - sv_maxclients = 30
debug(3) 22:12:55 qstat.c:4774 process_func_ret() - 0x565537f91550, 1
debug(3) 22:12:55 qstat.c:5658 cleanup_qserver() - cleanup_qserver 0x565537f91550, 0
debug(3) 22:12:55 qstat.c:5672 cleanup_qserver() - no server rules and next_player_info >= num_players, forcing close
debug(3) 22:12:55 qstat.c:5676 cleanup_qserver() - close_it 1
```
```json
[
	{
		"protocol": "q3s",
		"address": "gg.illwieckz.net:27960",
		"status": "online",
		"hostname": "gg.illwieckz.net:27960",
		"name": "gg.illwieckz.net { le frag courtois } backport",
		"gametype": "",
		"map": "perseus",
		"numplayers": 1,
		"maxplayers": 30,
		"numspectators": 0,
		"maxspectators": 0,
		"ping": 376,
		"retries": 0
	}
]
```
```sh
debug(2) 22:12:55 qstat.c:3245 do_work() - connected, post-packet_func: 0
debug(3) 22:12:55 qstat.c:4811 send_packets() - processing...
debug(3) 22:12:55 qstat.c:4922 send_packets() - done
debug(2) 22:12:55 qstat.c:3259 do_work() - connected: 0
```

Notice the `\bots\5` `\clients\1` fields in `infoResponse` packet, but there is 5 bots and 2 players, hence the server only reports one player (that's the wrong player count displayed in built-in server browser).

Now let's query a statusResponse instead:

```sh
qstat -d -d -d -json -R -P -q3s gg.illwieckz.net:27960
```
```sh
debug(1) 22:12:07 hcache.c:401 hcache_lookup_hostname() - looking up gg.illwieckz.net
debug(2) 22:12:07 hcache.c:412 hcache_lookup_hostname() - validating gg.illwieckz.net
debug(2) 22:12:07 hcache.c:417 hcache_lookup_hostname() - returning 3ed2472c
debug(1) 22:12:07 qstat.c:4546 bind_qserver2() - start 0x560ca6949560 @ 62.210.71.44:27960
debug(1) 22:12:07 qstat.c:4696 bind_sockets() - send 62.210.71.44:27960
debug(2) 22:12:07 qstat.c:4699 bind_sockets() - calling status_query_func for 0x560ca6949560 - connect
debug(3) 22:12:07 qstat.c:4774 process_func_ret() - 0x560ca6949560, 0
debug(3) 22:12:07 qstat.c:4811 send_packets() - processing...
debug(2) 22:12:07 qstat.c:3118 do_work() - connected: 1
debug(2) 22:12:07 qstat.c:5741 qserver_get_timeout() - timeout for 0x560ca6949560 is diff1 1000 diff2 65535
debug(2) 22:12:07 qstat.c:3138 do_work() - rc 1
debug(2) 22:12:07 qstat.c:3161 do_work() - recvfrom: 225
debug(1) 22:12:07 qstat.c:3183 do_work() - recv  50  50 62.210.71.44:27960
debug(2) 22:12:07 qstat.c:3192 do_work() - fill: 1 < 40
FROM gg.illwieckz.net:27960 len 225
    0: ffffffff 696e666f 52657370 6f6e7365  ....info Response
   16: 0a5c626f 74735c35 5c636c69 656e7473  .\bots\5 \clients
   32: 5c315c67 616d656e 616d655c 756e765c  \1\gamen ame\unv\
   48: 686f7374 6e616d65 5c5e3967 672e5e34  hostname \^9gg.^4
   64: 696c6c5e 37776965 5e31636b 7a5e392e  ill^7wie ^1ckz^9.
   80: 6e65745e 36207b20 5e356c65 20667261  net^6 {  ^5le fra
   96: 6720636f 7572746f 69735e36 207d5e39  g courto is^6 }^9
  112: 20626163 6b706f72 745e2a5c 6d61706e   backpor t^*\mapn
  128: 616d655c 70657273 6575735c 70726f74  ame\pers eus\prot
  144: 6f636f6c 5c38365c 70757265 5c315c73  ocol\86\ pure\1\s
  160: 65727665 726c6f61 645c385c 73746174  erverloa d\8\stat
  176: 735c6874 74703a2f 2f67672e 696c6c77  s\http:/ /gg.illw
  192: 6965636b 7a2e6e65 742f2373 74617473  ieckz.ne t/#stats
  208: 5c73765f 6d617863 6c69656e 74735c33  \sv_maxc lients\3
  224: 30                                   0

debug(2) 22:12:07 qstat.c:3243 do_work() - connected, pre-packet_func: 1
debug(2) 22:12:07 qstat.c:6133 deal_with_qw_packet() - deal_with_qw_packet 0x560ca6949560, 225
debug(2) 22:12:07 qstat.c:6429 deal_with_q2_packet() - deal_with_q2_packet 0x560ca6949560, 208
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - bots = 5
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: bots, value: 5, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - clients = 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - gamename = unv
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: gamename, value: unv, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - hostname = ^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - mapname = perseus
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - protocol = 86
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: protocol, value: 86, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - pure = 1
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: pure, value: 1, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - serverload = 8
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: serverload, value: 8, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - stats = http://gg.illwieckz.net/#stats
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: stats, value: http://gg.illwieckz.net/#stats, flags: 1
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_maxclients = 30
debug(3) 22:12:07 qstat.c:6199 deal_with_qw_packet() - send_rule_request_packet2
debug(3) 22:12:07 qstat.c:5535 send_rule_request_packet() - send_rule_request_packet: 0x560ca6949560
debug(3) 22:12:07 qstat.c:4774 process_func_ret() - 0x560ca6949560, 0
debug(2) 22:12:07 qstat.c:3245 do_work() - connected, post-packet_func: 1
debug(3) 22:12:07 qstat.c:4811 send_packets() - processing...
debug(2) 22:12:07 qstat.c:4845 send_packets() - server 0x560ca6949560, name ^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*, retry1 2, next_rule 0x560ca56db77a, next_player_info 65535, num_players 1, n_retries 3
debug(3) 22:12:07 qstat.c:4922 send_packets() - done
debug(2) 22:12:07 qstat.c:3259 do_work() - connected: 1
debug(2) 22:12:07 qstat.c:5741 qserver_get_timeout() - timeout for 0x560ca6949560 is diff1 950 diff2 65535
debug(2) 22:12:07 qstat.c:3138 do_work() - rc 1
debug(2) 22:12:07 qstat.c:3161 do_work() - recvfrom: 744
debug(1) 22:12:07 qstat.c:3183 do_work() - recv 122  72 62.210.71.44:27960
debug(2) 22:12:07 qstat.c:3192 do_work() - fill: 1 < 40
FROM gg.illwieckz.net:27960 len 744
    0: ffffffff 73746174 75735265 73706f6e  ....stat usRespon
   16: 73650a5c 425c2d2d 622d2d2d 2d626262  se.\B\-- b----bbb
   32: 622d2d2d 2d2d2d2d 2d2d2d2d 2d2d2d2d  b------- --------
   48: 2d2d2d2d 2d5c505c 3230322d 2d2d2d32  -----\P\ 202----2
   64: 3131312d 2d2d2d2d 2d2d2d2d 2d2d2d2d  111----- --------
   80: 2d2d2d2d 2d2d2d5c 675f6672 69656e64  -------\ g_friend
   96: 6c794275 696c6461 626c6546 6972655c  lyBuilda bleFire\
  112: 315c675f 66726965 6e646c79 46697265  1\g_frie ndlyFire
  128: 5c315c67 5f6d6178 47616d65 436c6965  \1\g_max GameClie
  144: 6e74735c 305c675f 6d6f6d65 6e74756d  nts\0\g_ momentum
  160: 48616c66 4c696665 5c355c67 5f6e6565  HalfLife \5\g_nee
  176: 64706173 735c305c 675f706f 77657252  dpass\0\ g_powerR
  192: 65616374 6f725261 6e67655c 31303030  eactorRa nge\1000
  208: 5c675f70 6f776572 52657065 61746572  \g_power Repeater
  224: 52616e67 655c3530 305c675f 756e6c61  Range\50 0\g_unla
  240: 67676564 5c315c67 5f756e6c 6f636b61  gged\1\g _unlocka
  256: 626c654d 696e5469 6d655c33 305c6761  bleMinTi me\30\ga
  272: 6d656e61 6d655c62 6173655c 6d61706e  mename\b ase\mapn
  288: 616d655c 70657273 6575735c 70726f74  ame\pers eus\prot
  304: 6f636f6c 5c38365c 73765f66 6c6f6f64  ocol\86\ sv_flood
  320: 50726f74 6563745c 305c7376 5f686f73  Protect\ 0\sv_hos
  336: 746e616d 655c5e39 67672e5e 34696c6c  tname\^9 gg.^4ill
  352: 5e377769 655e3163 6b7a5e39 2e6e6574  ^7wie^1c kz^9.net
  368: 5e36207b 205e356c 65206672 61672063  ^6 { ^5l e frag c
  384: 6f757274 6f69735e 36207d5e 39206261  ourtois^ 6 }^9 ba
  400: 636b706f 72745e2a 5c73765f 6d617852  ckport^* \sv_maxR
  416: 6174655c 305c7376 5f6d6178 636c6965  ate\0\sv _maxclie
  432: 6e74735c 33315c73 765f7072 69766174  nts\31\s v_privat
  448: 65436c69 656e7473 5c315c73 765f7374  eClients \1\sv_st
  464: 61747355 524c5c68 7474703a 2f2f6767  atsURL\h ttp://gg
  480: 2e696c6c 77696563 6b7a2e6e 65742f23  .illwiec kz.net/#
  496: 73746174 735c7469 6d656c69 6d69745c  stats\ti melimit\
  512: 305c7665 7273696f 6e5c556e 76616e71  0\versio n\Unvanq
  528: 75697368 65642030 2e353020 4c696e75  uished 0 .50 Linu
  544: 78207838 365f3634 20417072 20313120  x x86_64  Apr 11 
  560: 32303136 0a302033 3920225e 34696c6c  2016.0 3 9 "^4ill
  576: 5e377769 655e3163 6b7a205b 6772616e  ^7wie^1c kz [gran
  592: 6765725d 220a3020 34302022 556e6e61  ger]".0  40 "Unna
  608: 6d656450 6c617965 72233122 0a323338  medPlaye r#1".238
  624: 20302022 5e395b62 6f745d5e 34204d61   0 "^9[b ot]^4 Ma
  640: 6c6f220a 32363220 3020225e 395b626f  lo".262  0 "^9[bo
  656: 745d5e34 20416465 6c696e22 0a313438  t]^4 Ade lin".148
  672: 20302022 5e395b62 6f745d5e 31205572   0 "^9[b ot]^1 Ur
  688: 726f747a 220a3133 34203020 225e395b  rotz".13 4 0 "^9[
  704: 626f745d 5e31204d 6f726761 6e220a31  bot]^1 M organ".1
  720: 31382030 20225e39 5b626f74 5d5e3120  18 0 "^9 [bot]^1 
  736: 4e6f6c77 656e220a                    Nolwen". 

debug(2) 22:12:07 qstat.c:3243 do_work() - connected, pre-packet_func: 1
debug(2) 22:12:07 qstat.c:6133 deal_with_qw_packet() - deal_with_qw_packet 0x560ca6949560, 744
debug(2) 22:12:07 qstat.c:6429 deal_with_q2_packet() - deal_with_q2_packet 0x560ca6949560, 725
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - B = --b----bbbb--------------------
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: B, value: --b----bbbb--------------------, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - P = 202----2111--------------------
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: P, value: 202----2111--------------------, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_friendlyBuildableFire = 1
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_friendlyBuildableFire, value: 1, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_friendlyFire = 1
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_friendlyFire, value: 1, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_maxGameClients = 0
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_maxGameClients, value: 0, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_momentumHalfLife = 5
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_momentumHalfLife, value: 5, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_needpass = 0
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_needpass, value: 0, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_powerReactorRange = 1000
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_powerReactorRange, value: 1000, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_powerRepeaterRange = 500
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_powerRepeaterRange, value: 500, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_unlagged = 1
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_unlagged, value: 1, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - g_unlockableMinTime = 30
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: g_unlockableMinTime, value: 30, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - gamename = base
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: gamename, value: base, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - mapname = perseus
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - protocol = 86
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: protocol, value: 86, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_floodProtect = 0
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: sv_floodProtect, value: 0, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_hostname = ^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: sv_hostname, value: ^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_maxRate = 0
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: sv_maxRate, value: 0, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_maxclients = 31
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_privateClients = 1
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: sv_privateClients, value: 1, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - sv_statsURL = http://gg.illwieckz.net/#stats
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: sv_statsURL, value: http://gg.illwieckz.net/#stats, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - timelimit = 0
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: timelimit, value: 0, flags: 3
debug(3) 22:12:07 qstat.c:6460 deal_with_q2_packet() - version = Unvanquished 0.50 Linux x86_64 Apr 11 2016
debug(3) 22:12:07 qstat.c:7356 add_rule() - key: version, value: Unvanquished 0.50 Linux x86_64 Apr 11 2016, flags: 3
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:6505 deal_with_q2_packet() - player info
debug(3) 22:12:07 qstat.c:4774 process_func_ret() - 0x560ca6949560, 1
debug(3) 22:12:07 qstat.c:5658 cleanup_qserver() - cleanup_qserver 0x560ca6949560, 0
debug(3) 22:12:07 qstat.c:5672 cleanup_qserver() - no server rules and next_player_info >= num_players, forcing close
debug(3) 22:12:07 qstat.c:5676 cleanup_qserver() - close_it 1
```
```json
[
	{
		"protocol": "q3s",
		"address": "gg.illwieckz.net:27960",
		"status": "online",
		"hostname": "gg.illwieckz.net:27960",
		"name": "gg.illwieckz.net { le frag courtois } backport",
		"gametype": "",
		"map": "perseus",
		"numplayers": 7,
		"maxplayers": 31,
		"numspectators": 0,
		"maxspectators": 0,
		"ping": 50,
		"retries": 0,
		"rules": {
			"bots": "5",
			"gamename": "unv",
			"protocol": "86",
			"pure": "1",
			"serverload": "8",
			"stats": "http://gg.illwieckz.net/#stats",
			"B": "--b----bbbb--------------------",
			"P": "202----2111--------------------",
			"g_friendlyBuildableFire": "1",
			"g_friendlyFire": "1",
			"g_maxGameClients": "0",
			"g_momentumHalfLife": "5",
			"g_needpass": "0",
			"g_powerReactorRange": "1000",
			"g_powerRepeaterRange": "500",
			"g_unlagged": "1",
			"g_unlockableMinTime": "30",
			"sv_floodProtect": "0",
			"sv_hostname": "^9gg.^4ill^7wie^1ckz^9.net^6 { ^5le frag courtois^6 }^9 backport^*",
			"sv_maxRate": "0",
			"sv_privateClients": "1",
			"sv_statsURL": "http://gg.illwieckz.net/#stats",
			"timelimit": "0",
			"version": "Unvanquished 0.50 Linux x86_64 Apr 11 2016"
		},
		"players": [
			{
				"name": "illwieckz [granger]",
				"score": 0,
				"ping": 39
			},
			{
				"name": "UnnamedPlayer#1",
				"score": 0,
				"ping": 40
			},
			{
				"name": "[bot] Malo",
				"score": 238,
				"ping": 0
			},
			{
				"name": "[bot] Adelin",
				"score": 262,
				"ping": 0
			},
			{
				"name": "[bot] Urrotz",
				"score": 148,
				"ping": 0
			},
			{
				"name": "[bot] Morgan",
				"score": 134,
				"ping": 0
			},
			{
				"name": "[bot] Nolwen",
				"score": 118,
				"ping": 0
			}
		]
	}
]
```
```sh
debug(2) 22:12:07 qstat.c:3245 do_work() - connected, post-packet_func: 0
debug(3) 22:12:07 qstat.c:4811 send_packets() - processing...
debug(3) 22:12:07 qstat.c:4922 send_packets() - done
debug(2) 22:12:07 qstat.c:3259 do_work() - connected: 0
```

Since we get the complete player list, it's an alternate way to get the correct list, that's probably what does the [Servers](https://www.unvanquished.net/?page_id=340) page and why it's OK.

**Comment by illwieckz** [comment]
The bug is in [daemon/src/engine/server/sv_main.cpp](https://github.com/DaemonEngine/Daemon/blob/d0fdb3b59a8dc1980e6c9ed6c6222ff2f6644b7c/src/engine/server/sv_main.cpp#L574-L591):

```cpp

	// don't count privateclients
	int botCount = 0;
	int count = 0;

	for ( int i = sv_privateClients->integer; i < sv_maxclients->integer; i++ )
	{
		if ( svs.clients[ i ].state >= clientState_t::CS_CONNECTED )
		{
			if ( SV_IsBot(&svs.clients[ i ]) )
			{
				++botCount;
			}
			else
			{
				++count;
			}
		}
}
```

A private client is a client allowed to join even when the server is full. For example, if you set this in your server.cfg:

```
set sv_maxclients "31"          // number of players including private slots
set sv_privateclients "1"       // if set > 0, then this number of client slots will be reserved for connections
set sv_privatePassword "81G53CR37" // that have "password" set to the value of "sv_privatePassword"
```

The server will allow `31` slot but only `30` public slots, one people having set the `"81G53CR37"` can join even if the `30` slot are full, not not someone else. For example if your clan has 4 people and you set `sv_privateclients` to `4` and give the password to any member of your clan, you ensure your whole clan can join your server even if the 30 slots are already full.

So, so why the hell the private client number is always substracted from the client count. For example on my server I added one private slot to be sure I can join my server for administrative task. So, everytime someone browse the in-game server browser, my server is returning as player count, the player count minus myself, even if I'm not there. wat? :scream:

**Comment by Kangz** [comment]
Yes but what does the server browser show when 29 regular clients and one private client are connected:

 - 29/30 - This is confusing because there really are 30 players playing
 - 30/30 - This is confusing because the server isn't full yet.
 - 30/31 - This is confusing because it could also mean 30 regular clients, and that you cannot join unless you have the password.

**Comment by illwieckz** [comment]
Yes but it's still better than displaying a smaller amount of players in all case, in my case it was displaying an empty server everytime there was only one player… :-/

**Comment by illwieckz** [comment]
note that the fix is about returning a correct `client` value, it changes nothing about `sv_maxclients` being returned

we can also fake the `sv_maxclients` returned to substract the private clients. It would show 31/30 if private clients connect while the server is foll, but it looks to me, this extra one is privileged.

If we do not do that, people will be not able to join when the server says there is 31 slot available and only 30 players have joined…

**Comment by illwieckz** [comment]
in fact the code already does that:

```cpp
     info_map["sv_maxclients"] = std::to_string( sv_maxclients->integer - sv_privateClients->integer );
```

so if server 31 slot with 1 private, client receive 30 as maxium slot available, and if private player joins, it will displays 31/30.

**Comment by lavacano201014** [comment]
>Yes but what does the server browser show when 29 regular clients and one private client are connected:

* 29 + 1/30 - could (and early on, likely would) cause players to think that +1 is a bot, but clearly communicates to the player that there is one more slot open they could take.
* Change reporting to "number of slots free", 1/30 - also clear once you realize it's reporting the other way around, but no other game I know of does this, so that would likely make the learning curve worse (by how much, I couldn't say).
* Add new column, "number of slots open", to go alongside current player count - absolute least confusing, but likely requires the most work.

personally, if we change from the status quo, i'd like to vote for the first option

**Comment by illwieckz** [comment]
With the fix discussed, it shows like that (as the documentation in code expects):

- `4 + (4) / 8` → 8 slots, 0 private slot, 4 players, 4 bots
- `4 + (4) / 8` → 9 slots, 1 private slot not used, 4 players, 4 bots
- `8 + (1) / 8` → 9 slots, 1 private slot not used, 8 players, 1 bots
- `8 / 8` → 9 slots, 1 private slot not used, 8 players, 0 bots
- `9 / 8` → 9 slots, 1 private slot used, 9 players
- `8 / 8` → 8 slots, 0 private slot, 8 players, 0 bots
- `1 + (1) / 1` → 2 slots, 1 private slot used, 1 players, 1 bots
- `1 / 1` → 1 slots, 0 private slot, 1 players, 0 bots

Note: bot can't take a private slots.

So it always show to non-private player if he can join or not. Only the one who know the password™ know he can try to connect even if it's marked as full, the one who don't have password see it's full.

By the way I agree it would be cool to explicitely mark the server as full (not preventing players to try to connect though), not only relying on mental calculation. It does not need another field to read, it can be or special icon, a tag, a background color, graying out the text color, a visual style (striped) or something like that. We also lack a visual notice for password-only servers if I remember.

**Comment by slipher** [comment]
I prefer the numerator to be the number of players actually playing, and the denominator to be (number of public slots) + (number of in-use private slots). So the "capacity" of the server would increase by 1 when the server owner connects. (This might be what lavacano's second option is describing but I'm not sure). I think the private slot in use scenario is not important enough to increase the interface complexity by having 3 columns.

**Comment by DolceTriade** [comment]
+1 to @slipher's suggestion.

**Comment by illwieckz** [comment]
To do that, we need to get the number of connected private client to sum it [there](https://github.com/DaemonEngine/Daemon/blob/d0fdb3b59a8dc1980e6c9ed6c6222ff2f6644b7c/src/engine/server/sv_main.cpp#L637) before returning sv_maxclients field in infoResponse:

```cpp
Daemon/src/engine/server/sv_main.cpp:
	info_map["sv_maxclients"] = std::to_string( sv_maxclients->integer 
					- sv_privateClients->integer );
```

**Comment by lavacano201014** [comment]
@slipher no, that's not one of my suggestions, but it's probably more brilliant than all of mine

**Comment by illwieckz** [comment]
> To do that, we need to get the number of connected private client to sum it there before returning sv_maxclients field in infoResponse

Easier: returns client count instead if it's greater than non private max client count.

**Comment by illwieckz** [comment]
See my latest commit in DaemonEngine/Daemon#40

It hides private slot count when possible, but never returns a sv_maxclient number that is lower than publicly advertised player count.

**Comment by illwieckz** [comment]
The behavior of this new change:

- `4 + (4) / 8` → 8 slots, 0 private slot, 4 players, 4 bots
- `4 + (4) / 8` → 9 slots, 1 private slot not used, 4 players, 4 bots
- `8 + (1) / 9` → 9 slots, 1 private slot used , 8 players, 1 bots
- `8 / 8` → 9 slots, 1 private slot not used, 8 players, 0 bots
- `9 / 9` → 9 slots, 1 private slot used, 9 players
- `8 / 8` → 8 slots, 0 private slot, 8 players, 0 bots
- `1 + (1) / 2` → 2 slots, 1 private slot used, 1 players, 1 bots
- `1 / 1` → 1 slots, 0 private slot, 1 players, 0 bots

**Comment by illwieckz** [comment]
@Kangz when 29 regular clients and one private client are connected it prints 30/30 which is very ok and expected. The private slot is there to allow a private one to join even if the server is full, the private slot is not there to allow anyone to join because a private one has joined first.

Basically, if you set maxclients to 31 and 1 private slot, it's because you want the server being full at 30 but keep a backdoor to still be able to join. So it's wrong if it's become a way for non private people to join when there is already 30 people in server just because you're in.

As a server owner I add an extra private slot to allow myself to override the limit I've set, not to allow anyone to override the limit I've set because I'm there. Such kind of limits are set because of hardware limit (the server is known to lag with an higher count of people) or for gameplay purpose (the game is known to not be fun above that number of people), so it's very ok if only the private guy is the only one who is allowed to override the limit.

And well, if non-private guys would benefit from private perk, it would defeat the whole purpose of a private mechanism.
